### PR TITLE
Di 21181 abi

### PR DESCRIPTION
**Pull request by michaelatias**

### CODE REVIEW

File: client-side/src/app/views-and-editors/table/table.component.ts [hunk 1]
```diff
@@ -171,19 +174,19 @@ export class TableComponent{
...
   handleExportAction(){
     this.dimxService?.export({
       DIMXExportFormat: 'json',
       DIMXExportIncludeDeleted: false,
       DIMXExportFileName: EXPORT,
     })
   }
+  handleImportAction(){
+    this.dimxService?.import({
+      OverwriteObject: true,
```
Comment (roiaharon-pepperi) [inline client-side/src/app/views-and-editors/table/table.component.ts:186]: didn't you need to change it so the import won't overwrite on existing items?

File: client-side/src/app/views-and-editors/table/table.component.ts [hunk 2]
```diff
@@ -269,4 +272,23 @@ export class TableComponent{
...
     this.init()
   }
+  onOpenABI(){
+    this.addonBlockService.loadAddonBlockInDialog({
+      container: this.viewContainerRef,
+      name: 'ResourceSelection',
+      hostObject: {
+        resource: "ResourceHolder",
+        selectionMode: 'single',
+        view: "6a22f8ee-86f9-484b-8a66-e05b163f34b0"
```
Comment (roiaharon-pepperi) [inline client-side/src/app/views-and-editors/table/table.component.ts:282]: what is this view ID? why is it hard coded? please get it from a configuration or something

File: server-side/installation.ts [hunk 3]
```diff
@@ -81,37 +83,55 @@ async function createDIMXRelation(client: Client){
...
-    try {
-        const service = new AddonService(client)
-        const result = await Promise.all([
-            service.upsertRelation(exportViewsRelation),
-            service.upsertRelation(exportEditorsRelation)])
-        return  { success:true, resultObject: {result}};
-    }catch(e){
-        return { success: false, resultObject: e , errorMessage: `Error in upsert relation. error - ${e}`};
-    }
+// async function createDIMXRelation(client: Client){
```
Comment (roiaharon-pepperi) [inline server-side/installation.ts:86]: remove comment

**Review by roiaharon-pepperi — CHANGES_REQUESTED** [review]

File: client-side/src/app/resource-selection/resource-selection.component.ts [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+  constructor(
+    private viewsService: ViewsService,
+    private utilitiesService: UtilitiesService
+  ) { }
+
+  ngOnInit(): void {
+    this.init()
+  }
+  async init(){
+    const resultObject = await this.createGenericViewerConfiguration()
```
[the remark this answers is not in the document]
Comment (michaelatias) [reply client-side/src/app/resource-selection/resource-selection.component.ts:27]: will be fix in the next pull request

File: client-side/src/app/metadata.ts [hunk 5]
```diff
@@ -73,8 +73,27 @@ export interface IGenericViewerConfigurationObject{
...
+  allowNone?: boolean
+}
+
 export const SELECTION_TYPE = "SelectionType"
 export const EXPORT = "Export"
-export const IMPORT = "Import"
\ No newline at end of file
+export const IMPORT = "Import"
+
+type SussesObject<T> = {
```
[the remark this answers is not in the document]
Comment (michaelatias) [reply client-side/src/app/metadata.ts:89]: will be fixed in the next pr

[the remark this answers is not in the document]
Comment (michaelatias) [reply client-side/src/app/metadata.ts:89]: removed

File: client-side/src/app/metadata.ts [hunk 6]
```diff
@@ -73,8 +73,27 @@ export interface IGenericViewerConfigurationObject{
     selection: 'single' | 'multi'
   }
 }
-export const SELECTION_LIST = "SelectionList"
-export const DROP_DOWN = "DropDown"
+
+export interface ResourcePicker {
```
[the remark this answers is not in the document]
Comment (michaelatias) [reply client-side/src/app/metadata.ts:77]: removed

File: client-side/src/app/views-and-editors/table/table.component.ts (hunk 1 above)
Comment (michaelatias) [reply client-side/src/app/views-and-editors/table/table.component.ts:186]: yes you right

File: server-side/installation.ts (hunk 3 above)
Comment (michaelatias) [reply server-side/installation.ts:86]: removed

File: client-side/src/app/resource-selection/resource-selection.component.ts [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+    }
+    if(view.Resource.Name != this.hostObject.resource){
+      resultObject.error = 'ViewNotOfThatResourceError'
+      return resultObject
+    }
+    if(hostObject.selectionMode != 'single' && hostObject.selectionMode != 'multi'){
+      resultObject.error = 'IllegalSelectionModeError'
+      return resultObject
+    }
+    try{
```
[the remark this answers is not in the document]
Comment (michaelatias) [reply client-side/src/app/resource-selection/resource-selection.component.ts:79]: that function will be removed in the next pr

File: client-side/src/app/views-and-editors/table/table.component.ts (hunk 2 above)
Comment (michaelatias) [reply client-side/src/app/views-and-editors/table/table.component.ts:282]: that was for debugging only, i will keep it in comment

File: client-side/src/app/resource-selection/resource-selection.component.ts [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+      return
+    }
+    this.configurationObj = resultObject.data
+    this.loadCompleted = true
+  }
+  async  createGenericViewerConfiguration(): Promise<ResultObject<IGenericViewerConfigurationObject>>{
+    const resource = this.hostObject.resource
+    const resultOfView = await this.getView(resource, this.hostObject.view)
+    if(resultOfView.error){
+      resultOfView
```
[the remark this answers is not in the document]
Comment (michaelatias) [reply client-side/src/app/resource-selection/resource-selection.component.ts:40]: yes, removed

**Review by roiaharon-pepperi — APPROVED** [review]

**Pull request merged**